pkg/core/storelimit/store_limit.go

```diff
@@ -128,12 +129,15 @@ func (l *StoreRateLimit) Reset(rate float64, typ Type) {
 
 // limit the operators of a store
 type limit struct {
-	limiter    *ratelimit.RateLimiter
-	ratePerSec float64
+	limiter         *ratelimit.RateLimiter
+	ratePerSecMutex syncutil.RWMutex
+	ratePerSec      float64
 }
 
 // Reset resets the rate limit.
 func (l *limit) Reset(ratePerSec float64) {
+	l.ratePerSecMutex.Lock()
```

Comment:
Will it happen in the current code or just in your testing code?

Reply:
I think datarace can happen in the current code, we call `getOrCreateStoreLimit` in many places now, like in `AddOperator` or `tryAddOperators`, where we call `ExceedStoreLimit` to check the limit.

If the `getOrCreateStoreLimit` enters the `ResetStoreLimit` branch, then datarace is possible.

Reply:
And just like the test I built, this scenario is also possible to datarace if we set storelimit when other code calls `getOrCreateStoreLimit`.

Reply:
seems it is caused by https://github.com/tikv/pd/pull/8032

Reply:
> seems it is caused by #8032

sure, this test will fail since #8032